archive/imo/imo2006_q3.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2021 Tian Chen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Tian Chen
+-/
+
+import analysis.special_functions.sqrt
+
+/-!
+# IMO 2006 Q3
+
+Determine the least real number $M$ such that
+$$
+\left| ab(a^2 - b^2) + bc(b^2 - c^2) + ca(c^2 - a^2) \right|
+≤ M (a^2 + b^2 + c^2)^2
+$$
+for all real numbers $a$, $b$, $c$.
+
+## Solution
+
+The answer is $M = \frac{9 \sqrt 2}{32}$.
+
+This is essentially a translation of the solution in
+https://web.evanchen.cc/exams/IMO-2006-notes.pdf.
+
+It involves making the substitution
+`x = a - b`, `y = b - c`, `z = c - a`, `s = a + b + c`.
+-/
+
+open real
+
+/-- Replacing `x` and `y` with their average increases the left side. -/
+lemma rhs₁ {x y : ℝ} (hxy : 0 ≤ x * y) :
```

Comment:
Did you mean to call this `lhs₁`? That's what the docstring suggests.

Reply:
I think I just named them all `rhs`. I've changed them to be more descriptive